src/components/CourseChecklist/index.jsx

```diff
@@ -173,6 +173,11 @@ class CourseChecklist extends React.Component {
     this.state.checks.map((check) => {
       const isCompleted = this.isCheckCompleted(check.id);
       const shouldShowCommentSection = this.shouldShowCommentSection(check.id);
+      const shouldThisShowCheck = this.shouldShowCheck(check.id);
+
+      if (!shouldThisShowCheck) {
```

Comment:
You could also make the condition positive by checking

```
if (shouldShowCheck) {
  return (...);
} else {
  return null;
}
```

That might be easier to read.